From ca93ee5a495e169a413c75871cf4e92ad45fd706 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:38 -0400
Subject: [usb] fix coccinelle warnings

Message-id: <1410887463-20674-207-git-send-email-dzickus@redhat.com>
Patchwork-id: 95253
O-Subject: [RHEL7 PATCH 206/591] USB: fix coccinelle warnings
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 2e30d14f12421f39329faefd5ae92b02cc9ff255
Author: Fengguang Wu <fengguang.wu@intel.com>
Date:   Wed Dec 4 20:36:31 2013 -0800

    USB: fix coccinelle warnings

    drivers/usb/host/ehci-mv.c:181:26-27: WARNING comparing pointer to 0, suggest !E
    /c/kernel-tests/src/cocci/drivers/usb/host/ehci-mv.c:181:26-27: WARNING comparing pointer to 0

     Compare pointer-typed values to NULL rather than 0

    Semantic patch information:
     This makes an effort to choose between !x and x == NULL.  !x is used
     if it has previously been used with the function used to initialize x.
     This relies on type information.  More type information can be obtained
     using the option -all_includes and the option -I to specify an
     include path.

    Generated by: coccinelle/null/badzero.cocci

    CC: Jingoo Han <jg1.han@samsung.com>
    CC: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Fengguang Wu <fengguang.wu@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ehci-mv.c b/drivers/usb/host/ehci-mv.c
index 6516d24..474783f 100644
--- a/drivers/usb/host/ehci-mv.c
+++ b/drivers/usb/host/ehci-mv.c
@@ -178,7 +178,7 @@ static int mv_ehci_probe(struct platform_device *pdev)
 
  ehci_mv->phy_regs = devm_ioremap(&pdev->dev, r->start,
       resource_size(r));
- if (ehci_mv->phy_regs == 0) {
+ if (!ehci_mv->phy_regs) {
   dev_err(&pdev->dev, "failed to map phy I/O memory\n");
   retval = -EFAULT;
   goto err_clear_drvdata;
-- 
1.7.1